src/lib/sqlite/sqliterg.ts

```diff
@@ -0,0 +1,447 @@
+import type { SqliteRemoteDatabase } from 'drizzle-orm/sqlite-proxy'
+import { drizzle } from 'drizzle-orm/sqlite-proxy'
+import type { Invalidator, Readable, Subscriber, Unsubscriber, Writable } from 'svelte/store'
+import { get, writable } from 'svelte/store'
+import type { MigrationState, MigrationQueryMap, QueriesStringMap } from './migration'
+import { defaultMigrationQueryMap, defaultQueriesStringMap } from './migration'
+import type { GlobalState, Snippet, SnippetStore } from '$lib/utitlities/persistence'
+import {
+  clearTags,
+  clearAllTags as clearAllTags_,
+  deleteAllSnippets,
+  deleteSnippet as deleteSnippet_,
+  deleteSnippetsByFolder,
+  querySnippetsByFolderId,
+  upsertSnippet,
+  upsertTags, queryTagsBySnippetIds,
+} from './queries'
+import { buildTagsMap, dbSnippetToDisplaySnippet, displaySnippetToDbSnippet } from '$lib/utitlities/data-transformation'
+
+export type Params = {[key: string]: any}
+
+export type Transaction =
+  { noFail?: boolean } &
+  (
+    { query: string } |
+    { statement: string }
+  ) &
+  (
+    { values?: Params | any[] } |
+    { valuesBatch?: Params[] }
+  )
+
+export type Request = {
+  credentials?: {
+    user: string
+    password: string
+  }
+  transaction: Transaction[]
+}
+
+export type ResultTrue =
+  { success: true } &
+  (
+    { resultSet: any[] } |
+    { rowsUpdated: number } |
+    { rowsUpdatedBatch: number[] }
+  )
+
+export type ResultFalse = {
+  success: false
+  error: string
+}
+
+export type ResponseExecuteError = {
+  reqIdx: number
+  message: string
+}
+
+export type Result = ResultTrue | ResultFalse
+
+export type Response =
+  {
+    results: Result[]
+  } |
+  ResponseExecuteError
+
+export type SqlitergExecutor = {
+  isReachable(): Promise<boolean>
+  isAuthenticated(): Promise<boolean>
```

Comment:
These two functions are "overwhelming" the console with failed `fetch` calls. We should definitely look into that.

![image](https://github.com/thanhnguyen2187/crypta/assets/27758849/0766674d-31f0-4e2c-8923-8f77e562d0d4)